From 88f9a63d96bb947e6300ed2da4031015e99298fa Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 27 Jun 2022 10:54:15 +0300
Subject: [PATCH 17/33] firmware: scmi: Avoid fdtdec library

Use dev interface instead of fdtdec library to
avoid issues when using OF_LIVE.

Issue: ALB-9010
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/firmware/scmi.c | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/drivers/firmware/scmi.c b/drivers/firmware/scmi.c
index 0f61673cd2..05ef90abc4 100644
--- a/drivers/firmware/scmi.c
+++ b/drivers/firmware/scmi.c
@@ -93,17 +93,15 @@ struct scmi_smt_header {
 #define SMT_HEADER_MESSAGE_ID(id)	((id) & GENMASK(7, 0))
 
 struct scmi_shm_buf {
-	u8 *buf;
+	void __iomem *buf;
 	size_t size;
 };
 
 static int get_shm_buffer(struct udevice *dev, struct scmi_shm_buf *shm)
 {
-	int rc, len;
+	int rc;
 	struct ofnode_phandle_args args;
 	struct resource resource;
-	const fdt32_t *faddr;
-	phys_addr_t paddr;
 
 	rc = dev_read_phandle_with_args(dev, "shmem", NULL, 0, 0, &args);
 	if (rc)
@@ -113,20 +111,13 @@ static int get_shm_buffer(struct udevice *dev, struct scmi_shm_buf *shm)
 	if (rc)
 		return rc;
 
-	faddr = fdt_getprop(gd->fdt_blob, ofnode_to_offset(args.node),
-			    "reg", &len);
-	if (!faddr)
-		return len;
-
-	paddr = ofnode_translate_address(args.node, faddr);
-
 	shm->size = resource_size(&resource);
 	if (shm->size < sizeof(struct scmi_smt_header)) {
 		dev_err(dev, "Shared memory buffer too small\n");
 		return -EINVAL;
 	}
 
-	shm->buf = devm_ioremap(dev, paddr, shm->size);
+	shm->buf = devm_ioremap(dev, resource.start, shm->size);
 	if (!shm->buf)
 		return -ENOMEM;
 
-- 
2.17.1

